File: PL025.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

bool isPerfectArray(vector<int> &arr)
{
    for (int i = 0, j = arr.size() - 1; i < j; i++, j--)
    {
        if (arr[i] != arr[j])
        {
            return false;
        }
    }

    return true;
}

int main()
{
    vector<int> v = {1, 2, 3, 2, 1};
    cout << "Is array perfect or not : " << isPerfectArray(v) << endl;
    return 0;
}

File: PL025.java
public class PL025 {
    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 22, 1 };
        System.out.println("Is given array perfect : " + isPerfectArray(arr));

    }

    public static boolean isPerfectArray(int[] arr) {
        for (int i = 0, j = arr.length - 1; i < j; i++, j--) {
            if (arr[i] != arr[j]) {
                return false;
            }
        }

        return true;
    }
}
